src/irmin-pack/s.ml

```diff
@@ -74,6 +74,62 @@ module type S = sig
 
   val stats :
     dump_blob_paths_to:string option -> commit:commit -> repo -> unit Lwt.t
+
+  module Snapshot : sig
+    type kinded_hash = Contents of hash * metadata | Node of hash
+    [@@deriving irmin]
+
+    type entry = { name : string; hash : kinded_hash } [@@deriving irmin]
+
+    type inode_tree = {
+      depth : int;
+      length : int;
+      pointers : (int * hash) list;
+    }
+    [@@deriving irmin]
+
+    type v = Inode_tree of inode_tree | Inode_value of entry list
+    [@@deriving irmin]
+
+    type inode = { v : v; root : bool } [@@deriving irmin]
+
+    type t = [ `Inode of inode | `Blob of Backend.Contents.Val.t ]
```

Comment:
In `type t`, you mix polyvars and regular variants here. Would you change `t` to regular variants ?

PS: No need to change `on_disk` below

Reply:
thanks, done